blockchain/v2/processor.go

```diff
@@ -160,26 +155,32 @@ func (state *pcState) handle(event Event) (Event, error) {
 		firstPartsHeader := firstParts.Header()
 		firstID := types.BlockID{Hash: first.Hash(), PartsHeader: firstPartsHeader}
 
-		err = state.context.verifyCommit(state.chainID, firstID, first.Height, second.LastCommit)
+		err = state.context.verifyCommit(tdState.ChainID, firstID, first.Height, second.LastCommit)
 		if err != nil {
-			return pcBlockVerificationFailure{peerID: firstItem.peerID, height: first.Height}, nil
+			state.purgePeer(firstItem.peerID)
```

Comment:
We might want to add a note here that this policy might lead to liveness issue. We discussed once a possibility to analyse blocks in this case and end up either in case that we can identify who is faulty or that we have evidence for misbehaviour.

Reply:
What would you recommend as a note. Is there some issue we could link to?